model/riscv_pmp_regs.sail

```diff
@@ -154,15 +163,30 @@ function pmpReadCfgReg(n) = {
 }
 
 /* Helpers to handle locked entries */
-function pmpLocked(cfg: Pmpcfg_ent) -> bool =
-   cfg.L() == 0b1
+function pmpLockBit(cfg: Pmpcfg_ent) -> bool = cfg.L() == 0b1
+
+function pmpLocked(cfg: Pmpcfg_ent) -> bool = (cfg.L() == 0b1) & (mseccfg.RLB() == 0b0)
 
-function pmpTORLocked(cfg: Pmpcfg_ent) -> bool =
-   (cfg.L() == 0b1) & (pmpAddrMatchType_of_bits(cfg.A()) == TOR)
+function pmpTORLocked(cfg: Pmpcfg_ent) -> bool = (cfg.L() == 0b1) & (mseccfg.RLB() == 0b0) & (pmpAddrMatchType_of_bits(cfg.A()) == TOR)
 
-function pmpWriteCfg(cfg: Pmpcfg_ent, v: bits(8)) -> Pmpcfg_ent =
-  if pmpLocked(cfg) then cfg
-  else Mk_Pmpcfg_ent(v & 0x9f)  // Bits 5 and 6 are zero.
+function pmpWriteCfg(cfg: Pmpcfg_ent, v: bits(8)) -> Pmpcfg_ent = {
+  let legal_v : Pmpcfg_ent = Mk_Pmpcfg_ent(v & 0x9f);                    /* Constructing legal Pmpcfg_ent by making bit 5 and 6 zero */
+  let legal_v =
+    if ((mseccfg.MML() == 0b0) & (legal_v.R() == 0b0) & (legal_v.W() == 0b1))
+    then update_W(legal_v, 0b0)
+    else legal_v;
+  if pmpLocked(cfg) then cfg                                             /* If locked then configuration is unchanged */
+  else {
+    if (mseccfg.MML() == 0b1 & mseccfg.RLB() == 0b0 & legal_v.L() == 0b1) then {
+      if ((legal_v.R() == 0b0 & legal_v.W() == 0b0 & legal_v.X() == 0b1) |
+      (legal_v.R() == 0b0 & legal_v.W() == 0b1 & legal_v.X() == 0b0) |
```

Comment:
Indent continuation lines

Reply:
Why not use pattern matching. It's a subjective preference and I don't want to argue for my preference, but I find that complex nested conditionals are harder to read than pattern matching.

Reply:
You're right, that could well make more sense in this instance actually